server/auth/auth.go

```diff
@@ -135,6 +135,22 @@ func (p *SpiffeProvider) Authenticate(token string, r *http.Request) (knox.Princ
 	return NewService(splits[0], splits[1]), nil
 }
 
+// SpiffeFallbackProvider is a SpiffeProvider that uses the same Type byte as the
+// MTLSAuthProvider. The use case for this is to allow a client that specifies
+// MTLSAuth to also transparently be given Spiffe based access as well. For
+// more predictable results, ensure that the MTLSAuthProvider is registered before
+// the SpiffeFallbackProvider so that MTLSAuthProvider is always used if it succeeds.
+// Note that this is only possible with the SpiffeProvider because there is no use
+// of the token from the AuthorizationHeader in this Provider.
+type SpiffeFallbackProvider struct {
+	SpiffeProvider
+}
+
+// Type is set to be identical to the Type of the MTLSAuthProvider
+func (s *SpiffeFallbackProvider) Type() {
```

Comment:
TravisCI is complaining about the type here. you need to specify the return type.

Reply:
Sigh. Got too used to Python. :/